### DISCUSSION TITLE
SIPS w/ Client Cert

### DISCUSSION THREAD

**Discussion by ajmontero** [sipsorcery-org/sipsorcery#542] [Q&A] [answered]
I'm looking at the examples and it isn't obvious how I would go about creating a SIPS connection using a client certificate. Is that even possible?

I have tried adding the Client certificate when creating a TLS channel but that doesn't seem to send the cert in response to the server certificate in the handshake. Does anyone have an example of that working?

**Comment by sipsorcery** [comment, marked as answer]
Client certificates aren't supported. The main reason being it's never come up before. The SIP specification requires clients to be authenticated with an HTTP Digest. There's probably an extension for authenticating clients with X509 certificates but I'd be surprised if it's got much traction with SIP servers given the difficulties of managing X509 certificates on SIP devices like IP Phones, ATAs etc.

If you're interested in mondifying the sipsorcery library the spot where client certificates can be sent is [here](https://github.com/sipsorcery-org/sipsorcery/blob/533e127bc7873d087ba8ac713c4438859b75873f/src/core/SIP/Channels/SIPTLSChannel.cs#L117).
